### DISCUSSION TITLE
Add a light variant

### DISCUSSION THREAD

**Discussion by jochemvogel** [tailwindlabs/tailwindcss#6724] [Ideas]
Currently Tailwind includes a `dark` variant that lets you style your site differently when dark mode is enabled (https://tailwindcss.com/docs/dark-mode).

Most of the time I start with a dark theme and later on I'll implement a non-dark/light mode. So I was thinking: how 'bout a `light` variant.

I'm not sure whether this can be implemented at all. I was just thinking about it and I'll dump my idea here. 🙂

It's possible to prefix all the previous styling with `dark:` and add some light styling, so I'm not having any issues at all.

**Comment by alexanderbluhm** [comment]
Hi @jochemvogel 
You can look in the current Tailwind source code and you will see how the dark mode is implemented and adapt it to your light _variant_.
Pretty easy: https://github.com/tailwindlabs/tailwindcss/blob/master/src/corePlugins.js#L150

For example if you want the class syntax you can add a custom variant like this: https://play.tailwindcss.com/4lo8lmfkcm

**Comment by adamwathan** [comment]
Yep as @alexanderbluhm mentioned pretty easy to do this in user-land:

```js
// tailwind.config.js
const plugin = require('tailwindcss/plugin')

module.exports = {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    plugin(function ({ addVariant }) {
      // Class strategy
      addVariant('light', '.light &')

      // Media strategy
      addVariant('light', '@media (prefers-color-scheme: light)')
    }),
  ],
}
```

Not opposed to exploring adding it to core either, but this should at least let you do what you want regardless 👍🏻

**Comment by jochemvogel** [comment]
Thanks both! Will make development a bit easier for me. Didn't know that [variants](https://tailwindcss.com/docs/plugins#adding-variants) were a thing.
